test/hotspot/jtreg/compiler/integerArithmetic/TestNegMultiply.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2021, Red Hat, Inc. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug 8273454
+ * @summary Test transformation (-a)*(-b) = a*b
+ *
+ * @run main/othervm -XX:-TieredCompilation -XX:-BackgroundCompilation -XX:-UseOnStackReplacement TestNegMultiply
+ *
+ */
+
+import java.util.Random;
+
+public class TestNegMultiply {
+    private static Random random = new Random();
+    private static final int TEST_COUNT = 2000;
+
+    private static int test(int a, int b) {
+        return (-a) * (-b);
+    }
+
+    private static void testInt(int a, int b) {
+        int expected = (-a) * (-b);
```

Comment:
Are you sure about this is the expected value? As the method has been invoked 2000 times, I think it would be compiled by c2.

Reply:
The default CompileThreshold is 10K when tiered compilation is disabled, which is the case here, so there is no risk.

Reply:
I would prefer to keep as it is to match testxxx() functions. I think it articulates that JIT-ed result matches interpreter's.